client/spa/js/learning-resource/learning-resource.view.js

```diff
@@ -34,75 +60,53 @@ module.exports = Backbone.View.extend({
   },
 
   cancel: function(){
-    this.$('#title').val(this.model.get('title'));
-    this.$('#desc').val(this.model.get('description'));
-    this.$('#auth').val(this.model.get('authors'));
-    this.$('select option[value="'+this.model.get('resourceType')+'"]')
-        .attr('selected','selected');
     $('#form-area').removeClass('sty-form'); //remove bg color of form
-    $('#msg').empty()
-             .addClass('alert-warning')
-             .html('Changes cancelled')
-             .fadeIn().delay(2000).fadeOut('slow')
-             .queue(function(remove){
-                $('#msg').removeClass('alert-warning');
-                remove();
-             });
+    showMessage('alert-warning', 'Changes cancelled');
     this.$el.removeClass('editing');
   },
 
   save: function(){
     var view = this;
-    var auth = [];
-    if (this.$('#auth').val() === '') auth = null;
-    else $.each(this.$('#auth').val().split(','), function(){
-      auth.push($.trim(this));
-    });
-    var attributes = {
-      title: this.$('#title').val().trim(),
-      resourceType: $('#resourceType option:selected').val(),
-      description: this.$('#desc').val().trim(),
-      authors: auth
+
+    var authorsFormat = function(self) {
+      var auth = [];
+      if (self.$('#auth').val() === '') auth = null;
+      else $.each(self.$('#auth').val().split(','), function(){
+        auth.push($.trim(self));
```

Comment:
You are trimming self, which is an object. That is why you are getting [object object] in your array. The function gets a value parameter, that is what you want to trim and push into the array.